src/domain/comic/data/create.ts

```diff
@@ -1,14 +1,17 @@
 
 import database, { RecordData } from '^/integrations/database/module';
+import { generateId } from '^/integrations/utilities/crypto';
 
 import { RECORD_TYPE } from '../definitions/constants';
 import ComicData from './ComicData';
 
 export default async function create(imageId: string, structure?: string): Promise<ComicData>
 {
-    const record: RecordData = { imageId, structure };
+    const id = generateId();
 
-    const id = await database.createRecord(RECORD_TYPE, record);
+    const record: RecordData = { id, imageId, structure };
+
+    await database.createRecord(RECORD_TYPE, record);
 
     return new ComicData(id, imageId, structure);
```

Comment:
```suggestion
    return mapRecord(record);
```

With the addition of the `id`, the record is complete and can be used for mapping. Saves on double work.